dask/dataframe/io/hdf.py

```diff
@@ -473,6 +473,20 @@ def read_hdf(
         paths = sorted(glob(pattern))
     else:
         paths = pattern
+
+    if not isinstance(pattern, str) and len(paths) == 0:
+        raise ValueError("No files provided")
+    if not paths or len(paths) == 0:
+        raise IOError("File(s) not found: {0}".format(pattern))
+    for path in paths:
+        try:
+            exists = os.path.exists(path)
+        except (ValueError, TypeError):
+            exists = False
```

Comment:
Under what circumstances does `os.path.exists` raise these types of exceptions?

If it doesn't raise exceptions then I wonder if this might be simpler as ...

```python
for path in paths:
    if not os.path.exists(path):
        raise IOError(...)
```

Reply:
`ValueError` is raised if for example the file name contains an embedded null byte. One example raising this exception is `os.path.exists('a\x00b')`, found at <https://bugs.python.org/issue33721>.
`TypeError` is raised if the argument is of nonsensical type, like a list (`os.path,exists([])` does this).
So while those exceptions can be raised in general, it's a good question if they can be raised here. I think it mostly depends on how `stringify_path` treats such weird cases, but I couldn't find any in-depth documentation of this function.